src/components/TodoForm/TodoForm.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useState } from 'react';
+
+import usersFromServer from '../../api/users';
+import { Todo, TypeError } from '../../types';
+import { getMaxNumberId } from '../../utils/getMaxNumberId';
+import { getUserById } from '../../utils/getUserById';
+
+type Props = {
+  onAddTodo: (newElementTodo: Todo) => void;
+  todo: Todo[];
+};
+
+export const TodoForm: React.FC<Props> = ({ onAddTodo, todo }) => {
+  const [titleError, setTitleError] = useState<string>(TypeError.DEFAULT);
+  const [title, setTitle] = useState('');
+
+  const [selectedUserId, setSelectedUserId] = useState(0);
+  const [selectedUserIdError, setSelectedUserIdError] = useState<string>(
+    TypeError.DEFAULT,
+  );
+
+  const handleUserInput = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setTitleError(TypeError.DEFAULT);
+  };
+
+  const handleSelectedInput = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectedUserId(+event.target.value);
+    setSelectedUserIdError(TypeError.DEFAULT);
+  };
+
+  const reset = () => {
+    setTitle('');
+    setSelectedUserId(0);
+    setTitleError(TypeError.DEFAULT);
+    setSelectedUserIdError(TypeError.DEFAULT);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!title) {
+      setTitleError(TypeError.TITLE_ERROR);
+    }
+
+    if (!selectedUserId) {
+      setSelectedUserIdError(TypeError.USER_ERROR);
+    }
+
+    if (!title || !selectedUserId) {
+      return;
+    }
+
+    const newTodo: Todo = {
+      id: getMaxNumberId(todo),
+      title: title.trim(),
+      completed: false,
+      userId: selectedUserId,
+      user: getUserById(selectedUserId),
```

Comment:
[CODE KNOWLEDGE] - When creating a new todo item, it is unnecessary to include the user object within the todo object here. This can lead to data duplication and potential inconsistencies. Instead, consider storing only the userId and retrieving the user details separately when needed.